fissa/extraction.py

```diff
@@ -133,9 +132,24 @@ def image2array(image):
             A 3D array containing the data, with dimensions corresponding to
             ``(frames, y_coordinate, x_coordinate)``.
         """
-        if isinstance(image, basestring):
-            return tifffile.imread(image)
-        return np.array(image)
+        if not isinstance(image, basestring):
+            return np.array(image)
+
+        with tifffile.TiffFile(image) as tif:
+            frames = []
+            for page in tif.pages:
+                page = page.asarray()
+                if page.ndim < 2:
+                    raise EnvironmentError(
+                        "TIFF {} has pages with {} dimensions (page shaped {})."
+                        " Pages must have at least 2 dimensions".format(
+                            image, page.ndim, page.shape
+                        )
+                    )
+                shp = [-1] + list(page.shape[-2:])
```

Comment:
We should perhaps throw a warning at least when the dimensionality is higher than 3, and state that we concatenate all dimensions > 3.

Reply:
Probably a good idea, yes.

Reply:
I've added the warnings, and tests that make sure the warnings are warned.

However, it turns out most higher-dimensional TIFF stack really are loaded frame-by-frame when loaded page-by-page with TiffReader, so we can't always tell that the structure it was saved as had more than 3 dimensions. How I've done it is the best we can do though.